cmake/thirdparty/get_legate_core.cmake

```diff
@@ -0,0 +1,63 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_or_configure_legate_core)
+  set(oneValueArgs VERSION REPOSITORY BRANCH EXCLUDE_FROM_ALL)
+  cmake_parse_arguments(PKG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(FIND_PKG_ARGS      ${PKG_VERSION}
+      GLOBAL_TARGETS     legate::core
+      BUILD_EXPORT_SET   cunumeric-exports
+      INSTALL_EXPORT_SET cunumeric-exports)
+
+  # First try to find legate_core via find_package()
+  # so the `Legion_USE_*` variables are visible
+  rapids_find_package(legate_core ${FIND_PKG_ARGS} QUIET)
+
+  if(legate_core_FOUND)
+    message(STATUS "CPM: using local package legate_core@${PKG_VERSION}")
+  else()
+    include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/cpm_helpers.cmake)
+    get_cpm_git_args(legate_core_cpm_git_args REPOSITORY ${PKG_REPOSITORY} BRANCH ${PKG_BRANCH})
+    rapids_cpm_find(legate_core ${FIND_PKG_ARGS}
+        CPM_ARGS
+          ${legate_core_cpm_git_args}
```

Comment:
This seems to have changed from previous versions:

```
  rapids_cpm_find(legate_core ${PKG_VERSION}
      GLOBAL_TARGETS          legate::core
      BUILD_EXPORT_SET        cunumeric-exports
      INSTALL_EXPORT_SET      cunumeric-exports
      CPM_ARGS
        GIT_REPOSITORY        ${PKG_REPOSITORY}
        GIT_TAG               ${PKG_PINNED_TAG}
        EXCLUDE_FROM_ALL      ${PKG_EXCLUDE_FROM_ALL}
        OPTIONS               "Legion_USE_CUDA ${Legion_USE_CUDA}"
                              "Legion_USE_OpenMP ${OpenMP_FOUND}"
                              "Legion_USE_GASNet ${Legion_USE_GASNet}"
                              "Legion_BOUNDS_CHECKS ${CUNUMERIC_CHECK_BOUNDS}"
  )
```

See the note above on naming of CMake variables. I have a slight preference for the CMake option being `Cunumeric_USE_CUDA` and then forwarding that value to `Legion_USE_CUDA` here so that all option names have a common prefix and we're not naming flags after dependencies. Not a strong preference.